src/wh_client_nvm.c

```diff
@@ -833,51 +838,47 @@ int wh_Client_NvmAddObjectDma64(whClientContext* c,
     }
     return rc;
 }
+#endif /* WH_DMA_IS_64BIT */
 
 /** NVM AddObjectDMA Helper functions */
 int wh_Client_NvmAddObjectDmaRequest(whClientContext* c,
         whNvmMetadata* metadata,
         whNvmSize data_len, const uint8_t* data)
 {
-    if (sizeof(intptr_t) == sizeof(uint32_t)) {
-        return wh_Client_NvmAddObjectDma32Request(c,
-                (intptr_t)metadata, data_len, (intptr_t)data);
-    }
-    if (sizeof(intptr_t) == sizeof(uint64_t)) {
-        return wh_Client_NvmAddObjectDma64Request(c,
-                (intptr_t)metadata, data_len, (intptr_t)data);
-    }
+#if WH_DMA_IS_32BIT
+    return wh_Client_NvmAddObjectDma32Request(c, (intptr_t)metadata, data_len,
```

Comment:
Consider simply #defining NvmAddObjectDmaRequest to be NAODR32 or 64 and eliminate this function?

Reply:
My thoughts were that this makes it an official API and gets it doxygen documentation. Any reasonable compiler should optimize out the double function call too. But LMK if you would rather it be a macro redirection, I'm happy to change it (and some of the other ones like the NVM DMA read, as well as copy to/from client)